### PR TITLE
[ConfigBuilder] Replace all framework config

### PR DESCRIPTION
**Pull request by Nyholm**
This PR will replace all framework configuration with the new ConfigBuilders.

### CODE REVIEW

**Comment by Nyholm** [comment]
Only the config reference and messeger left =)

**Comment by Nyholm** [comment]
I cannot figure out why not all errors show in the diff.. I only see the first 5..

File: configuration/env_var_processors.rst [hunk 1]
```diff
@@ -269,10 +277,13 @@ Symfony provides the following env var processors:
...
-            $container->setParameter('env(TRUSTED_HOSTS)', '["10.0.0.1", "10.0.0.2"]');
-            $container->loadFromExtension('framework', [
-                'trusted_hosts' => '%env(json:TRUSTED_HOSTS)%',
-            ]);
+            use Symfony\Component\DependencyInjection\ContainerBuilder;
+            use Symfony\Config\FrameworkConfig;
+
+            return static function (ContainerBuilder $container, FrameworkConfig $framework) {
+                $container->setParameter('env(TRUSTED_HOSTS)', '["10.0.0.1", "10.0.0.2"]');
+                $framework->trustedHosts('%env(json:TRUSTED_HOSTS)%');
```
Comment (Nyholm) [inline configuration/env_var_processors.rst]: This does not work. `$framework->trustedHosts()` expects an array.

File: configuration/env_var_processors.rst [hunk 2]
```diff
@@ -353,10 +364,13 @@ Symfony provides the following env var processors:
...
-            $container->setParameter('env(TRUSTED_HOSTS)', '10.0.0.1,10.0.0.2');
-            $container->loadFromExtension('framework', [
-                'trusted_hosts' => '%env(csv:TRUSTED_HOSTS)%',
-            ]);
+            use Symfony\Component\DependencyInjection\ContainerBuilder;
+            use Symfony\Config\FrameworkConfig;
+
+            return static function (ContainerBuilder $container, FrameworkConfig $framework) {
+                $container->setParameter('env(TRUSTED_HOSTS)', '10.0.0.1,10.0.0.2');
+                $framework->trustedHosts('%env(csv:TRUSTED_HOSTS)%');
```
Comment (Nyholm) [inline configuration/env_var_processors.rst]: Same here.

File: lock.rst [hunk 3]
```diff
@@ -269,12 +273,16 @@ provides :ref:`named lock <reference-lock-resources-name>`:
...
-                'invoice' => ['semaphore', 'redis://r2.docker'],
-                'report' => 'semaphore',
-            ],
-        ]);
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $framework) {
+            $framework->lock()
+                ->resource('lock', [
+                    'invoice' => ['semaphore', 'redis://r2.docker'],
```
Comment (Nyholm) [inline lock.rst]: We cannot use an array here... Hm..

> Invalid type for path "framework.lock.resources.lock.invoice". Expected "scalar", but got "array".

Comment (Nyholm) [reply lock.rst]: @jderusse Can you look at this comment specifically?

File: configuration/env_var_processors.rst (hunk 1 above)
Comment (Nyholm) [reply configuration/env_var_processors.rst]: @nicolas-grekas Should we make any changes for this scenario?

File: lock.rst (hunk 3 above)
Comment (jderusse) [reply lock.rst]: I think this should be
```
$framework->lock()
                ->resource('invoice', ['semaphore', 'redis://r2.docker'])
                ->resource('cron', 'flock')
```

File: configuration/env_var_processors.rst (hunk 1 above)
Comment (nicolas-grekas) [reply configuration/env_var_processors.rst]: Hmm. Should we remove type hints?
Alternatively, should we accept array|EnvConfigurator?

**Comment by Nyholm** [comment]
Hm... About the CI not showing all errors. 

It will not show [this error](https://github.com/symfony/symfony-docs/pull/15269/checks#step:11:49) and anything after it. Earlier, it did not show [this error](https://github.com/symfony/symfony-docs/runs/2403843016#step:11:11) and anything after it. 

It make sense if there were a special character that was being printed, but I dont see any. But all `::error` [commands](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message) are "reset" after line break. So if one has wrong syntax then it wont effect the next line. 

(Btw I [urlencode the line breaks](https://github.com/symfony-docs-tools/code-block-checker/blob/main/src/Command/CheckDocsCommand.php#L181) to have a multi line comment.)

File: lock.rst (hunk 3 above)
Comment (Nyholm) [reply lock.rst]: Oh, yes. I misunderstood the original config. Thank you

File: configuration/env_var_processors.rst (hunk 1 above)
Comment (Nyholm) [reply configuration/env_var_processors.rst]: Hm.. that means removing type hints for all "leaf" values and add them in the doc block. 

```diff
    /**
+    * @param bool|EnvConfigurator $value
     * @default false
     * @return $this
     */
-   public function enabled(bool $value): self
+   public function enabled($value): self
    {
        $this->enabled = $value;
    
        return $this;
    }
```

File: lock.rst (hunk 3 above)
Comment (jderusse) [reply lock.rst]: beware, maybe the second case is not allowed by the PHP contract? because it should be an array

Comment (Nyholm) [reply lock.rst]: Yes. You are correct. I updated the PR to use an array.

File: configuration/env_var_processors.rst (hunk 1 above)
Comment (nicolas-grekas) [reply configuration/env_var_processors.rst]: Correct. I don't see a better way. We could generate union types on SF6.

Comment (Nyholm) [reply configuration/env_var_processors.rst]: Same. 
I'll make a PR to Symfony then. 

> We could generate union types on SF6.

Yes =)

Comment (jderusse) [reply configuration/env_var_processors.rst]: That's the same issue with data converted by normalizers everywhere

Probably the same issue as https://github.com/symfony/symfony-docs/pull/15269#discussion_r618154902

Comment (jderusse) [reply configuration/env_var_processors.rst]: > We could generate union types on SF6

Thast mean, we should be able to detect the allowed types. That could be doable with `beforeNormalization()->ifString` but not for `beforeNormalization()->ifTrue`

Comment (Nyholm) [reply configuration/env_var_processors.rst]: Hm.. Im not sure it is the same as the "lock issue". This is related to env vars. 

See https://github.com/symfony/symfony/pull/40903

File: lock.rst [hunk 4]
```diff
@@ -129,32 +129,34 @@ this behavior by using the ``lock`` key like:
...
-            'lock' => [
-                'invoice' => ['semaphore', 'redis://r2.docker'],
-                'report' => 'semaphore',
-            ],
-        ]);
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $framework) {
+            $framework->lock()
+                ->resource('lock', null)
```
Comment (jderusse) [inline lock.rst]: This should be `'default'` which is the internal value used when the user did not provide named resource

File: lock.rst [hunk 5]
```diff
@@ -129,32 +129,34 @@ this behavior by using the ``lock`` key like:
...
+                ->resource('lock', ['pgsql+advisory:host=127.0.0.1;dbname=lock'])
+                ->resource('lock', ['sqlsrv:server=127.0.0.1;Database=app'])
+                ->resource('lock', ['oci:host=127.0.0.1;dbname=app'])
+                ->resource('lock', ['mongodb://127.0.0.1/app?collection=lock'])
+                ->resource('lock', ['%env(LOCK_DSN)%'])
+
+                // named locks
+                ->resource('invoice', ['semaphore', 'redis://r2.docker'])
+                ->resource('report', ['semaphore'])
+                ;
```
Comment (jderusse) [inline lock.rst]:
```suggestion
            ;
```
for consistency

File: lock.rst (hunk 4 above)
Comment (Nyholm) [reply lock.rst]: Is this what you mean?

```suggestion
                ->resource('lock', ['default'])
```

Comment (jderusse) [reply lock.rst]:
```suggestion
                ->resource('default', null)
                ->resource('default', ['flock'])
```

In the previous version, the word `'lock'` was for `framework.lock`, which is now replaced by `$framework->lock()`

Comment (jderusse) [reply lock.rst]: hmm I'm not sure that `null` is allowed with this new syntax

Comment (Nyholm) [reply lock.rst]: Technically, no =)

It violates the contract by PHP doc block. But there is no complaint as long as the Config Definition supports it.

File: messenger.rst [hunk 6]
```diff
@@ -1643,23 +1627,21 @@ this globally (or for each transport) to a service that implements
...
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $framework) {
+            $messenger = $framework->messenger();
+
+            $messenger->serializer()
+                ->defaultSerializer('messenger.transport.symfony_serializer')
+                ->symfonySerializer()
+                    ->format('json')
+                    ->context('foo', 'bar');
```
Comment (Nyholm) [inline messenger.rst:1639]: Do we have a better example than this?

**Review by Nyholm — COMMENTED** [review]
Assuming https://github.com/symfony/symfony/pull/40903 will be merged. This PR is ready for a final review.

**Comment by Nyholm** [comment]
Wohoo. The code PR is merged.

**Comment by Nyholm** [comment]
Is there something I can do to help this PR to get merged?

**Comment by javiereguiluz** [comment]
Sorry it took us some time to merge this. Thanks Tobias!

**Comment by Nyholm** [comment]
Thank you for merging. I'll fix the rest now

**Pull request merged**